frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -118,28 +115,31 @@ const InstallNeedsApprovalMessage: React.FC<InstallNeedsApprovalMessageProps> =
   </>
 );
 
-const CreateCRDButton: React.FC<CRDButtonProps> = ({ obj, disabled }) => {
-  if (!getCRDFromObject(obj)) {
+export const CreateInitializationResourceButton: React.FC<InitializationResourceButtonProps> = ({
+  obj,
+  disabled,
+  targetNamespace,
+}) => {
+  if (!getInitializationResourceJSON(obj)) {
     return null;
   }
-  let cr = null;
+  let initializationResource = null;
   try {
-    cr = JSON.parse(getCRDFromObject(obj));
+    initializationResource = JSON.parse(getInitializationResourceJSON(obj));
   } catch (error) {
     const errorMsg = error.message;
     errorModal({ errorMsg });
   }
-  const crNamespace = cr?.[0]?.metadata.namespace;
-  const apiVersion = cr?.[0]?.apiVersion;
-  const kind = cr?.[0]?.kind;
-  const { group, version } = groupVersionFor(apiVersion);
-  const reference = referenceForGroupVersionKind(group)(version)(kind);
+  const initializationResourceNamespace =
+    initializationResource?.metadata?.namespace || targetNamespace;
```

Comment:
We should be checking `model.namespaced`. (There's some code duplication here, which makes things harder.)